link/kprobe.go

```diff
@@ -294,3 +303,37 @@ func kprobePrefix(ret bool) string {
 	}
 	return "p"
 }
+
+// determineRetprobeBit reads a Performance Monitoring Unit's retprobe bit
+// from /sys/bus/event_source/devices/<pmu>/format/retprobe.
+func determineRetprobeBit(pmu string) (uint64, error) {
+	parseFunc := func(pmu string) (uint64, error) {
+		p := filepath.Join("/sys/bus/event_source/devices/", pmu, "/format/retprobe")
+
+		data, err := ioutil.ReadFile(p)
+		if err != nil {
+			return 0, err
+		}
+
+		var rp uint64
+		n, err := fmt.Sscanf(string(bytes.TrimSpace(data)), "config:%d", &rp)
+		if err != nil {
+			return 0, fmt.Errorf("parse retprobe bit: %w", err)
+		}
+		if n != 1 {
+			return 0, fmt.Errorf("parse retprobe bit: expected 1 item, got %d", n)
+		}
+
+		return rp, nil
+	}
+
+	switch pmu {
+	case "kprobe":
```

Comment:
Instead of this, we could provide two wrapper functions, `kretprobeBit()` and `uretprobeBit()`, that wrap `determineRetprobeBit()` in their respective `sync.Once`s.